docs/RFCs/private-network.md

```diff
@@ -0,0 +1,66 @@
+- Feature Name: private-network
+- Start Date: 2022-02-01
+- RFC PR:
+- Functionland Issue:
+
+## Background
+we are using IPFS as our file system. but IPFS is build to use for public data, and it does not support ACL, 
+so we need to find a way to keep user safe until our security layer become mature.
+
+## Problem Statement
+We need to protect user and his/her data from harms and risk of public network.  
+
+## Motivation
+Isolating User from public network can help us relax the problem and build useful product and testing our security layer without putting user in harms way. 
+
+## Proposal
+using IPFS build-in private network. base on [spec](https://github.com/libp2p/specs/blob/master/pnet/Private-Networks-PSK-V1.md)
+Its uses a pre-shared key (PSK) to create a private network with encrypted communication.
+
+For box setup user provide an environment variable SECRET which is password he should remember.
+the secret then convert to a hash of 256 bit by algorithm like sha256 and generate the swarm.key for ipfs and libp2p node's.
+
+For Fula client when user call createClient he should also provide the password he used for setting the box's.
```

Comment:
I think by adding the human readable password i make this PR a little bit complicated. but my thinking was asking user to enter a 64  character in web app or photo's is painful task so i suggest to  change it to more human readable think.

Reply:
I will try to separate them.

Reply:
I think when we can use DID instead of Peer id to connect (verified connection) box <---> box and client <---> box which also most cover internal ipfs protocols like bitsawp engine then that's the end if life for this proposal

Reply:
> I think when we can use DID instead of Peer id to connect (verified connection) box <---> box and client <---> box which also most cover internal IPFS like bitsawp engine then that's the end of life for this proposal

like the libp2p gatekeeper should use the DID to let others connect to my box or ... .

Reply:
and if we want to use ipfs-cluster then it also have to cover a lot of ground on go lang too.